lib/Conversion/TTIRToTTNN/TTIRToTTNN.cpp

```diff
@@ -134,6 +134,16 @@ class ToLayoutOpConversionPattern
 public:
   using OpConversionPattern<ttir::ToLayoutOp>::OpConversionPattern;
 
+  bool shouldForceRowMajor(ttir::ToLayoutOp op) const {
+    for (mlir::Operation *user : op.getResult().getUsers()) {
+      if (isa<ttir::Conv2dOp>(user) || isa<ttir::MaxPool2dOp>(user)) {
```

Comment:
Will there ever be a case where the consumer ops include unary/binary ops alongside conv/pool ops? In that case would forcing row_major be an issue here?

Reply:
You're right! If there are multiple users of a tensor, e.g. graph forks into both an op that expects tile layout and and op that expects row major layout, that'd be an issue here.

@sdjordjevicTT you have the most visibility, has this been talked about earlier?

Reply:
I'll try to post a PR for #880 tomorrow, I could squeeze in a fix for this as well.

Reply:
Alright, this is a more general problem, I've opened #890 to track.